src/tableediting.js

```diff
@@ -0,0 +1,238 @@
+
+		// Table cell conversion.
+		conversion.for( 'editingDowncast' ).add( downcastInsertCell( { asWidget: true } ) );
+		conversion.for( 'dataDowncast' ).add( downcastInsertCell() );
+
+		conversion.for( 'upcast' ).add( upcastElementToElement( { model: 'tableCell', view: 'td' } ) );
+		conversion.for( 'upcast' ).add( upcastElementToElement( { model: 'tableCell', view: 'th' } ) );
+
+		// Table attributes conversion.
+		conversion.attributeToAttribute( { model: 'colspan', view: 'colspan' } );
+		conversion.attributeToAttribute( { model: 'rowspan', view: 'rowspan' } );
+
+		conversion.for( 'editingDowncast' ).add( downcastTableHeadingColumnsChange( { asWidget: true } ) );
+		conversion.for( 'dataDowncast' ).add( downcastTableHeadingColumnsChange() );
+		conversion.for( 'editingDowncast' ).add( downcastTableHeadingRowsChange( { asWidget: true } ) );
+		conversion.for( 'dataDowncast' ).add( downcastTableHeadingRowsChange() );
+
+		editor.commands.add( 'insertTable', new InsertTableCommand( editor ) );
+		editor.commands.add( 'insertRowAbove', new InsertRowCommand( editor, { order: 'above' } ) );
+		editor.commands.add( 'insertRowBelow', new InsertRowCommand( editor, { order: 'below' } ) );
+		editor.commands.add( 'insertColumnBefore', new InsertColumnCommand( editor, { order: 'before' } ) );
+		editor.commands.add( 'insertColumnAfter', new InsertColumnCommand( editor, { order: 'after' } ) );
+
+		editor.commands.add( 'removeRow', new RemoveRowCommand( editor ) );
+		editor.commands.add( 'removeColumn', new RemoveColumnCommand( editor ) );
+
+		editor.commands.add( 'splitCellVertically', new SplitCellCommand( editor, { direction: 'vertically' } ) );
+		editor.commands.add( 'splitCellHorizontally', new SplitCellCommand( editor, { direction: 'horizontally' } ) );
+
+		editor.commands.add( 'mergeCellRight', new MergeCellCommand( editor, { direction: 'right' } ) );
+		editor.commands.add( 'mergeCellLeft', new MergeCellCommand( editor, { direction: 'left' } ) );
+		editor.commands.add( 'mergeCellDown', new MergeCellCommand( editor, { direction: 'down' } ) );
+		editor.commands.add( 'mergeCellUp', new MergeCellCommand( editor, { direction: 'up' } ) );
+
+		editor.commands.add( 'setTableHeaders', new SetTableHeadersCommand( editor ) );
+
+		this.listenTo( editor.editing.view.document, 'keydown', ( ...args ) => this._handleTabOnSelectedTable( ...args ) );
+		this.listenTo( editor.editing.view.document, 'keydown', ( ...args ) => this._handleTabInsideTable( ...args ) );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ TableUtils ];
+	}
+
+	/**
+	 * Handles {@link module:engine/view/document~Document#event:keydown keydown} events for 'Tab' key executed
+	 * when table widget is selected.
+	 *
+	 * @private
+	 * @param {module:utils/eventinfo~EventInfo} eventInfo
+	 * @param {module:engine/view/observer/domeventdata~DomEventData} domEventData
+	 */
+	_handleTabOnSelectedTable( eventInfo, domEventData ) {
```

Comment:
AFAICS, <kbd>Shift + Tab</kbd> moves selection to the first cell as well. How about moving it to the last cell in that case? Then, you could traverse the table from the end using <kbd>Shift + Tab</kbd>. Looks to me like this could make some sense.